test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ClassTest.scala

```diff
@@ -44,15 +46,65 @@ class ClassTest {
   )
 
   @Test def getPrimitiveTypeName(): Unit = {
-    assertEquals("void", classOf[Unit].getName)
-    assertEquals("boolean", classOf[Boolean].getName)
-    assertEquals("char", classOf[Char].getName)
-    assertEquals("byte", classOf[Byte].getName)
-    assertEquals("short", classOf[Short].getName)
-    assertEquals("int", classOf[Int].getName)
-    assertEquals("long", classOf[Long].getName)
-    assertEquals("float", classOf[Float].getName)
-    assertEquals("double", classOf[Double].getName)
+    @noinline
+    def testNoInline(expected: String, cls: Class[_]): Unit =
+      assertEquals(expected, cls.getName())
+
+    @inline
+    def test(expected: String, cls: Class[_]): Unit = {
+      testNoInline(expected, cls)
+      assertEquals(expected, cls.getName())
+    }
+
+    test("void", classOf[Unit])
+    test("boolean", classOf[Boolean])
+    test("char", classOf[Char])
+    test("byte", classOf[Byte])
+    test("short", classOf[Short])
+    test("int", classOf[Int])
+    test("long", classOf[Long])
+    test("float", classOf[Float])
+    test("double", classOf[Double])
```

Comment:
We seem to be not testing that `classOf[Some[_]].getName` yield the right result in both optimized and non-optimized scenario. (there is `java_lang_Class_getName_under_normal_circumstances` above, but it does not have the inline / non-inline treatment).